wagtail/contrib/settings/templatetags/wagtailsettings_tags.py

```diff
@@ -0,0 +1,21 @@
+from django.template import Library
+
+from wagtail.wagtailcore.models import Site
+
+from ..context_processors import SettingsProxy
+
+register = Library()
+
+
+@register.simple_tag(takes_context=True)
+def get_settings(context, use_default_site=False):
```

Comment:
This could be implemented for `get_settings`, but I don't see the template tag being used much. It is only as a fallback in case you are in a strange situation. With the pending deprecation that @nealtodd pointed out, supporting assignment tags across multiple versions of Django will become quite annoying. I do not think it is worth the bother.